Added Null check on Representation Property

amitjoy opened the pull request:
`org.eclipse.smarthome.core.thing.type.ThingType.getRepresentationProperty()` javadoc says that representation property of this thing type can be null. Without any representation property results in NPE.

tomhoefer commented on bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessor.java:
```diff
@@ -79,13 +81,13 @@ public void receiveTypedEvent(ThingStatusInfoChangedEvent event) {
-                    logger.debug("Auto-ignoring the inbox entry for the representation value {}", value);
-                    inbox.setFlag(result.getThingUID(), DiscoveryResultFlag.IGNORED);
-                }
+            Thing thing = thingRegistry.stream()
```
I think it is worth to write a unit test for this purpose ....

amitjoy replied:
Done

triller-telekom commented:
Thanks for the fix, but while you are at it anyway, how about raising a warning in the IDE for the next one who uses `ThingType.getRepresentationProperty()`?

Just add a `@Nullable` in front of the return type `String` here: https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingType.java#L312

amitjoy commented:
@tomhoefer I have added a new unit test to check thing when it does not contain any representation property.

@triller-telekom I have added a new commit to introduce JDT null-check annotation in `ThingType#getRepresentationProperty()`
